src/datasources/asset/data-sources/AssetDataSourceBase.ts

```diff
@@ -45,6 +45,14 @@ export abstract class AssetDataSourceBase extends DataSourceBase<AssetQuery, Ass
     }
   }
 
+  getQueryParams(): URLSearchParams {
+    return new URLSearchParams(window.location.search);
```

Comment:
@saradei-ni I'd double check that grafana uses the same mechanism in their sourcecode in to get the variables. My only concern is to have a usecase where this code is not running in the browser.  Maybe it would be better to pass the query params as parameters to the datasource 

https://grafana.com/developers/plugin-tools/tutorials/build-a-data-source-backend-plugin

Reply:
the rest of variables are handled via query params as well
I added a way to handle already in time range case as well.